src/shared/features/AppletSettings/ExportDataSetting/ExportDataSetting.tsx

```diff
@@ -1,46 +1,149 @@
+    control,
+    inputSx: { width: '19rem' },
+  };
+
+  const onCloseCallback = () => {
+    if (toDate < fromDate) {
+      setValue('toDate', addDays(fromDate, 1));
+    }
+  };
+  const onDateTypeChange = (e: SelectEvent) => {
+    const dateType = e.target.value as ExportDateType;
+    switch (dateType) {
+      case ExportDateType.AllTime:
+        setValue('fromDate', minDate);
+        setValue('toDate', maxDate);
+        break;
+      case ExportDateType.Last24h:
+        setValue('fromDate', addDays(maxDate, -1));
+        setValue('toDate', maxDate);
+        break;
+      case ExportDateType.LastWeek:
+        setValue('fromDate', addDays(maxDate, -7));
+        setValue('toDate', maxDate);
+        break;
+      case ExportDateType.LastMonth:
+        setValue('fromDate', addDays(maxDate, -30));
+        setValue('toDate', maxDate);
+        break;
+      case ExportDateType.ChooseDates:
+        setValue('fromDate', minDate);
+        setValue('toDate', maxDate);
+        break;
+    }
+  };
 
   return (
-    <>
-      <StyledAppletSettingsDescription>{t('exportDescription')}</StyledAppletSettingsDescription>
-      <Box sx={{ width: 'fit-content' }}>
-        <StyledAppletSettingsButton
-          onClick={() => setDataIsExporting(true)}
-          variant="outlined"
-          startIcon={<Svg width="18" height="18" id="export" />}
-          data-testid={dataTestid}
-        >
-          {t('download')}
-        </StyledAppletSettingsButton>
-      </Box>
-      {dataIsExporting && (
-        <DataExportPopup
-          isAppletSetting
-          popupVisible={dataIsExporting}
-          setPopupVisible={setDataIsExporting}
-          chosenAppletData={appletData ?? null}
-          data-testid={`${dataTestid}-popup`}
+    <FormProvider {...methods}>
+      <form noValidate autoComplete="off">
+        <StyledAppletSettingsDescription>{t('exportDescription')}</StyledAppletSettingsDescription>
+        <SelectController
+          name={'dateType'}
+          control={control}
+          options={getDateTypeOptions()}
+          label={t('dateRange')}
+          data-testid={`${dataTestid}-dateType`}
+          dropdownStyles={{
+            width: '15rem',
+          }}
+          SelectProps={{
+            autoWidth: true,
+          }}
+          customChange={onDateTypeChange}
         />
-      )}
-    </>
+        {hasCustomDate && (
+          <StyledFlexTopCenter
+            sx={{
+              mt: theme.spacing(2.4),
+            }}
+          >
+            <DatePicker
+              {...commonProps}
+              name="fromDate"
+              onCloseCallback={onCloseCallback}
+              label={t('fromDate')}
+              data-testid={`${dataTestid}-from-date`}
+            />
+            <StyledBodyLarge sx={{ margin: theme.spacing(0, 0.8) }}>{t('smallTo')}</StyledBodyLarge>
+            <DatePicker
+              {...commonProps}
+              name="toDate"
+              minDate={fromDate}
```

Comment:
how about minDate={createdAt} for datepicker above? another question)

Reply:
I used it in `commonProps`, but later it was updated with different values. So I will add a separate property for datepicker above.